xunit-autolabeler-v2/ast_parser/python/invoker.py

```diff
@@ -45,12 +47,28 @@ def get_json_for_dir(root_dir: str) -> List[Dict]:
     test_files = [file for file in python_files
                   if constants.TEST_FILE_MARKER in file]
 
+    test_method_map = dict()
     for file in test_files:
-        _parse_test(file, source_methods)
+        tests = _parse_test(file, source_methods)
+        for test_keys, test_value in tests.items():
+            key_str = (
+                test_keys[0] + lib_constants.KEY_SEPARATOR + test_keys[1])
 
-    for method in source_methods:
-        new_source_path = os.path.relpath(
-            method.drift.source_path, root_dir)
-        method.drift._replace(source_path=new_source_path)
+            if key_str not in test_method_map:
+                test_method_map[key_str] = []
 
-    return [method.drift._asdict() for method in source_methods]
+            test_method_map[key_str] += test_value
+
+    for method in source_methods:
+        new_source_path = os.path.abspath(
+            method.drift.source_path)
+        method.drift.source_path = new_source_path
+
+    output = {
+        'snippets': [
+            dataclasses.asdict(method.drift) for method in source_methods
+        ],
+        'test_method_map': test_method_map
+    }
+
+    return output
```

Comment:
Just return inline if the variable is not going to be used elsewhere:

```
return {
  'snippets': foo,
  'test_method_map': bar
}
```

Reply:
Done.